src/plays/context-with-real-usecases/ContextWithRealUsecases.jsx

```diff
@@ -35,22 +35,18 @@ function ContextWithRealUsecases(props) {
 
   //prev slide
   const handlePrev = () => {
-    if (index > 1) {
-      setIndex(index - 1);
-    } else setIndex(5);
+    setIndex((preV) => (preV > 1 ? preV - 1 : 5));
   };
 
   //next slide
-  const handleNext = () => {
-    if (index < 5) {
-      setIndex(index + 1);
-    } else setIndex(1);
-  };
+  const handleNext = useCallback(() => {
```

Comment:
why would you need to use `useCallback`?

Reply:
@koustov , We are using this function in ueEffect and we have to add this function to the useEffect dependency. so on every render function reference should not get updated that's why we have to warp it with `useCallbak`

Reply:
I see the `index` value is already initialized to `0` then why would we need useEffect? And I see the `ContextInfoSlider` expecting to have this value initialized with `1`, then lets do it during the constant creation, I mean

```javascript
const [index, setIndex] = useState(1);
```

Reply:
@koustov , Here use case is totally different please see the whole file you will get why we are doing this.